effects/extrinsic.ts

```diff
@@ -161,3 +166,22 @@ function $extrinsic<
     .as<number>()
   return scale.$extrinsic(deriveCodec_, metadata_, sign!, addrPrefix)
 }
+
+function $call<
+  Client extends Z.$<rpc.Client> = Z.$<rpc.Client>,
+>(client: Client) {
+  const metadata_ = metadata(client)()
+  const deriveCodec_ = scale.deriveCodec(metadata_)
+  return scale.$call(deriveCodec_, metadata_)
+}
+
+export function callHash<
+  Client extends Z.$<rpc.Client> = Z.$<rpc.Client>,
+  Call extends Z.$<unknown> = Z.$<unknown>,
+>(client: Client) {
+  return (call: Call) => {
+    return Z.ls($call(client), call).next(([codec, call]) => {
+      return Blake2_256.hash(codec.encode(call))
+    })
+  }
+}
```

Comment:
@harrysolovay I'm not sure about the best location for these `$call` and `callHash`